site_scons/buildutils.py

```diff
@@ -34,6 +36,409 @@
+
+        return out
+
+    def help(self, env: "Optional[SCEnvironment]" = None) -> str:
+        """Convert option help for command line interface (CLI) output"""
+        # assemble output
+        out = f"{self._build_title(backticks=False, indent=2)}"
+        out += self._build_description(backticks=False, indent=4)
+        out += self._build_defaults(self.default, backticks=False, indent=2, hanging=2)
+
+        if env is None:
+            return out
+
+        # Get the actual value in the current environment
+        actual = env.subst(f"${self.name!s}") if self.name in env else None
+
+        # Fix the representation of Boolean options to match the default values
+        if actual in ["True", "False"]:
+            actual = True if actual == "True" else False
+
+        # Print the value if it differs from the default
+        if str(actual) != str(self.default):
+            out += self._build_defaults(
+                actual, backticks=False, title="actual", indent=2, hanging=2)
+
+        return out
+
+
+class PathOption(Option):
+    """Object corresponding to SCons PathVariable"""
+
+    def to_scons(self, env: "Optional[SCEnvironment]" = None) -> "SCVariables":
+        return PathVariable(*super().to_scons(env))
+
+
+class BoolOption(Option):
+    """Object corresponding to SCons BoolVariable"""
+
+    def to_scons(self, env: "Optional[SCEnvironment]" = None) -> "SCVariables":
+        return BoolVariable(*super().to_scons(env))
+
+
+class EnumOption(Option):
+    """Object corresponding to SCons EnumVariable"""
+
+    def to_scons(self, env: "Optional[SCEnvironment]" = None) -> "SCVariables":
+        return EnumVariable(*super().to_scons(env))
+
+
+class Configuration:
+    """
+    Class enabling selection of options based on a dictionary of `Option` objects
+    that allows for a differentiation between platform/compiler dependent options.
+
+    In addition, the class facilitiates the generation of formatted help text and
+    reST documentation.
+    """
+
+    header = [
+        textwrap.dedent(
+            """
+                **************************************************
+                *   Configuration options for building Cantera   *
+                **************************************************
+
+        The following options can be passed to SCons to customize the Cantera
+        build process. They should be given in the form:
+
+            scons build option1=value1 option2=value2
+
+        Variables set in this way will be stored in the 'cantera.conf' file and reused
+        automatically on subsequent invocations of scons. Alternatively, the
+        configuration options can be entered directly into 'cantera.conf' before
+        running 'scons build'. The format of this file is:
+
+            option1 = 'value1'
+            option2 = 'value2'
+        """
+        )
+    ]
+
+    def __init__(self):
+        self.options = {}
+        self.exported = []
```

Comment:
```suggestion
        self.options: "Dict[str, Union[str, bool, dict]]" = {}
        self.exported: "List[str]" = []
```
Ideally that `dict` in the `Union` would also be typed 😁

Reply:
Thought I had this one, but didn't. Now fixed.